src/main/java/de/tum/in/www1/artemis/service/learningpath/LearningPathService.java

```diff
@@ -280,4 +292,61 @@ public NgxLearningPathDTO generateNgxGraphRepresentation(@NotNull LearningPath l
     public NgxLearningPathDTO generateNgxPathRepresentation(@NotNull LearningPath learningPath) {
         return this.learningPathNgxService.generateNgxPathRepresentation(learningPath);
     }
+
+    /**
+     * Finds a learning path by its id and eagerly fetches the competencies, linked lecture units and exercises, and the corresponding domain objects storing the progress of the
+     * connected user.
+     * <p>
+     * As Spring Boot 3 doesn't support conditional JOIN FETCH statements, we have to retrieve the data manually.
+     *
+     * @param learningPathId the id of the learning path to fetch
+     * @return the learning path with fetched data
+     */
+    public LearningPath findWithCompetenciesAndLearningObjectsAndCompletedUsersById(long learningPathId) {
```

Comment:
This looks like a very complex and lengthy process. It includes at least 4 queries to the database, if I can see that correctly. I hope there is no query in a forEach method.

@rstief @MaximilianAnzinger could you please double check if all this data is actually needed and think about ways to make this faster?

Reply:
It is indeed rather complex and lengthy. To be precise: It includes exactly 4 queries. I made sure to first retrieve all relevant data from the database and then fill the return entity in the for loops.

If we don't need the data, that would obviously make things simpler, regarding the approach. This is sadly the best approach I could come up with after researching and testing myself and testing approaches suggested by others for at least a total of 13 hours. Entity Graphs, Specifications, different queries, other dependencies...
Though, if anyone has a specific idea that would improve this, I'd be glad ^^